components/dashboards-web-component/src/auth/utils/AuthManager.jsx

```diff
@@ -0,0 +1,151 @@
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+import AuthenticationAPI from '../../utils/apis/AuthenticationAPI';
+
+/**
+ * Name of the session cookie.
+ */
+const sessionUser = 'wso2dashboard_user';
+
+/**
+ * Authentication manager.
+ */
+export default class AuthManager {
+    /**
+     * Get user from the session cookie.
+     *
+     * @returns {{}|null} User object
+     */
+    static getUser() {
+        const buffer = AuthManager.getSessionCookie(sessionUser);
+        return buffer ? JSON.parse(buffer) : null;
+    }
+
+    /**
+     * Set user into a session cookie.
+     *
+     * @param {{}} user  User object
+     */
+    static setUser(user) {
+        AuthManager.setSessionCookie(sessionUser, JSON.stringify(user), 7 * 24 * 3600 * 1000);
+    }
+
+    /**
+     * Delete user from the session cookie.
+     */
+    static clearUser() {
+        AuthManager.deleteSessionCookie(sessionUser);
+    }
+
+    /**
+     * Check whether the user is logged in.
+     *
+     * @returns {boolean} Status
+     */
+    static isLoggedIn() {
+        return !!AuthManager.getUser();
+    }
+
+    /**
+     * Authenticate the user and set the user into the session.
+     *
+     * @param {string} username Username
+     * @param {string } password Password
+     * @param {boolean} rememberMe Remember me flag
+     * @returns {Promise} Promise
+     */
+    static authenticate(username, password, rememberMe) {
+        return new Promise((resolve, reject) => {
+            AuthenticationAPI
+                .login(username, password, rememberMe)
+                .then(() => {
+                    // TODO: Get user roles from the SCIM API.
+                    const roles = [];
+                    AuthManager.setUser({ username, rememberMe, roles });
+                    resolve();
+                })
+                .catch(error => reject(error));
+        });
+    }
+
+    /**
+     * Logout user by revoking tokens and clearing the session.
+     *
+     * @returns {Promise} Promise
+     */
+    static logout() {
+        return new Promise((resolve, reject) => {
+            AuthenticationAPI
+                .logout()
+                .then(() => {
+                    AuthManager.clearUser();
+                    resolve();
+                })
+                .catch(error => reject(error));
+        });
+    }
+
+    /**
+     * Set session cookie.
+     *
+     * @param {string} name Name of the cookie
+     * @param {string} value Value of the cookie
+     * @param {number} expiresIn Expires in
```

Comment:
Shall we mention the unit ? (ex : in miliseconds)

Reply:
+1